Userland/Libraries/LibGfx/ImageFormats/AVIFLoader.cpp

```diff
@@ -0,0 +1,211 @@
+        if (result != AVIF_RESULT_OK) {
+            avifDecoderDestroy(decoder);
+            avifRGBImageFreePixels(&rgb);
+            return Error::from_string_literal("Conversion from YUV failed");
+        }
+
+        // FIXME: support high bitdepth
+        VERIFY(rgb.depth == 8);
+        memcpy(bitmap->scanline(0), rgb.pixels, rgb.rowBytes * decoder->image->height);
+
+        context.frame_descriptors.append(ImageFrameDescriptor { bitmap, static_cast<int>(decoder->imageTiming.duration * 1000) });
+
+        context.state = AVIFLoadingContext::BitmapDecoded;
+    }
+
+    avifDecoderDestroy(decoder);
+    avifRGBImageFreePixels(&rgb);
+    return {};
+}
+
+bool AVIFImageDecoderPlugin::set_error(ErrorOr<void> const& error_or)
+{
+    if (error_or.is_error()) {
+        dbgln("AVIFImageDecoderPlugin error: {}", error_or.error());
+        m_context->state = AVIFLoadingContext::State::Error;
+        return true;
+    }
+    return false;
+}
+
+IntSize AVIFImageDecoderPlugin::size()
+{
+    return m_context->size.value();
+}
+
+bool AVIFImageDecoderPlugin::sniff(ReadonlyBytes data)
+{
+    AVIFLoadingContext context;
+    context.data = data;
+
+    // Intentionally no set_error() call: We're just sniffing `data` passed to the function, not reading m_context->data.
+    return !decode_avif_header(context).is_error();
+}
+
+ErrorOr<NonnullOwnPtr<ImageDecoderPlugin>> AVIFImageDecoderPlugin::create(ReadonlyBytes data)
+{
+    auto context = TRY(try_make<AVIFLoadingContext>());
+    auto plugin = TRY(adopt_nonnull_own_or_enomem(new (nothrow) AVIFImageDecoderPlugin(data, move(context))));
+    TRY(decode_avif_header(*plugin->m_context));
+    return plugin;
+}
+
+bool AVIFImageDecoderPlugin::is_animated()
+{
+    return m_context->image_count > 1;
+}
+
+size_t AVIFImageDecoderPlugin::loop_count()
+{
+    if (is_animated()) {
+        return m_context->repetition_count;
+    } else {
+        return 0;
+    }
+}
+
+size_t AVIFImageDecoderPlugin::frame_count()
+{
+    if (!is_animated())
+        return 1;
+    return m_context->image_count;
+}
+
+size_t AVIFImageDecoderPlugin::first_animated_frame_index()
+{
+    return 0;
+}
+
+ErrorOr<ImageFrameDescriptor> AVIFImageDecoderPlugin::frame(size_t index, Optional<IntSize>)
+{
+    if (index >= frame_count())
+        return Error::from_string_literal("AVIFImageDecoderPlugin: Invalid frame index");
+
+    if (m_context->state == AVIFLoadingContext::State::Error)
+        return Error::from_string_literal("AVIFImageDecoderPlugin: Decoding failed");
+
+    if (m_context->state < AVIFLoadingContext::State::BitmapDecoded) {
+        TRY(decode_avif_image(*m_context));
+
+        m_context->state = AVIFLoadingContext::State::BitmapDecoded;
+    }
+
+    return m_context->frame_descriptors[index];
+}
+
+ErrorOr<Optional<ReadonlyBytes>> AVIFImageDecoderPlugin::icc_data()
+{
+    return Error::from_string_literal("AVIFImageDecoderPlugin::icc_data NYI");
+}
```

Comment:
I think if it isn't implemented anyway, you can remove the override. The base method return an empty optional, which is fine.